parser.js

```diff
@@ -483,7 +483,22 @@
 
       var atom = parseAtomAndExtendedAtom();
       if (!atom) {
-        bail('Expected atom');
+        // Check if a quantifier is following. A quantifier without an atom
+        // is an error.
+        pos_backup = pos
+        var quantifier = parseQuantifier() || false;
+        if (quantifier) {
+          pos = pos_backup
+          bail('Expected atom');
+        }
+
+        // If no unicode flag, then try to parse ExtendedAtom -> ExtendedPatternCharacter.
+        //      ExtendedPatternCharacter
+        if (!hasUnicodeFlag && (res = matchReg(/^[^^$\\.*+?()[|]/))) {
```

Comment:
Aren't `^$\\.*([]` already consumed by `parseAtomAndExtendedAtom`, so we don't need to handle them here?

I think this should only be `!hasUnicodeFlag && current('{')`, since we only need to handle the `{...` case here.

Reply:
Thanks for the catch. You are right - it's not necessary to parse the full ExtendedPatternCharacter list.

 I've updated the code to `!hasUnicodeFlag && (res = matchReg(/^{/))`.